### PR TITLE
Track server-side invitation events with PostHog

### LINKED ISSUE #4381: Improve tracking of accepted invites
### Description

Currently, when a user receives a team join invitation, and has to create a FlowFuse user in the process, the user does not count as "accepting" the invitation in PostHog.

**Desired Behavior**
Any time a user receives a team join invitation, clicks the invitation, and ends up a part of a team as a result, that user should be counted as "accepting" the invitation in PostHog.

### Epic/Story

_No response_

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

### PR DESCRIPTION
**Pull request by joepavitt**
## Description

- Registers a new "product" service for the `app`
- Adds `posthog-node` as a new dependency
- Wraps the `capture` function for PostHog, which is initiated using the same `config` as the frontend
- Wire the product service into the accepting of team invites server-side
- Remove the `invite-accepted` event from client-side so as to not duplicate recording.

### Todo

- Unit tests?

## Related Issue(s)

Closes #4381 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass

### CODE REVIEW

**Comment by joepavitt** [comment]
Have a working end-to-end:

Screenshot 2024-08-16 at 15 20 36

**Review by Steve-Mcl — COMMENTED** [review]
looks good to me Joe but in the spirit of "done" there are no tests.

I think at minimum we should have unit tests that checks `app.product.capture` is a function. Better still would be to perform an invite, spy `app.notifications.send` and check the arguments in the call are sane (no raw `id`s)

**Comment by joepavitt** [comment]
Yep, haven't touched the server-side unit tests before, so wasn't sure how deep to go with it.

**Comment by joepavitt** [comment]
Not sure why the `notifications.send` is picked up in the diff, as that should be on `main` already

**Comment by joepavitt** [comment]
@Steve-Mcl I've updated the branch now so that the diff _doesn't_ include all the stuff from https://github.com/FlowFuse/flowfuse/pull/4385/files too

File: package.json [hunk 1]
```diff
@@ -95,6 +95,7 @@
         "pg": "^8.11.3",
         "pino": "^8.15.1",
         "pino-pretty": "^10.0.0",
+        "posthog-node": "^4.1.0",
```
Comment (Steve-Mcl) [inline package.json:98]: there was a guard for null values added in 4.1.1.  also V4.2.0 is released.

We should probably switch to 4.1.1 at min.

**Review by Steve-Mcl — CHANGES_REQUESTED** [review]

**Review by Steve-Mcl — APPROVED** [review]
good to go

**Pull request merged**